vnext/PropertySheets/Autolink.targets

```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- 
+  Copyright (c) Microsoft Corporation. All rights reserved.
+ Licensed under the MIT License.. 
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Target Name="RunAutolink" BeforeTargets="PrepareForBuild">
```

Comment:
What expected inputs? The commands and args are defined in the props file, it still works if there are no community modules found.

Reply:
> What expected inputs? The commands and args are defined in the props file, it still works if there are no community modules found.

@jonthysell in order for MSBuild to know that it's safe to not re-run a Target, it has to know what the inputs and outputs are. If the existing outputs are older than the inputs' last modified timestamp (and command line options haven't changed), then msbuild will not re-run a target. You'd specify the list of files that are inputs and outputs as an attribute to the Target.
See: https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-targets?view=vs-2019

Reply:
There are no input files, the only "input" is the result of the `react-native config` command (which is run within the autolinkin command), which I don't want to write to disk. And since this just does a check (with no files updated) there are no output file either.

Reply:
FYI @vmoroz since there was some recent discussion about our incremental builds, and cutting down on anything that we can't do incrementally. I'm okay with this for now personally though, if it's hard to make this incremental. @jonthysell do you know current built-time impact?

An important bit that I forgot to brought up is that Visual Studio normally does a dumb thing with build logic where it will skip running targets entirely if filter files don't change. I disable this right now in `ReactPatches.targets` by doing this:

```xml
  <PropertyGroup>
    <DisableFastUpToDateCheck>True</DisableFastUpToDateCheck>
  </PropertyGroup>
```

Removing that would break this as well, so we should consider putting that in a place more global if we are going to continue adding targets not derived from filter files.

Reply:
The autolinking check only occurs on app project builds and runs in ~500-600 milliseconds. ReactPatches.targets is only imported for our libraries, not app projects. Changes there won't have any impact on the app projects.

Reply:
`ReactPatches.targets` is still run as part of the MSRN build. The prop is weird and applies to the current Visual Studio instance instead of a specific project, so it currently leaks pretty far. You can test by removing the prop then restarting Visual Studio, but it definitely has an impact on user projects right now.

Reply:
The worse case scenario if this target doesn't run - if the dev changes their depdencies on community modules (add, remove, update) and hasn't run re-run auto-linking once (typically by running run-windows), they won't be warned that they should run the command. If their build breaks because that last time auto-linking, the generated files are now invalid, then they'll see an error in those generated files, and see the command they should run.

Reply:
I don't mean to keep harping on this, but we have concrete user scenarios where the wrong thing will happen. Could we have an conversation on any concerns preventing adding this single prop to something reachable from external projects? I understand there's workarounds for when this fails, but unless I'm missing something, the fix here should be pretty trivial.

There's been consensus from multiple engineers that IDE support is something that we do care about. Without DisableFastUpToDateCheck defined somewhere we won't run the target unless native build files change. This works now because the property affects the global state of Visual Studio, so it affecting internal projects also impacts MSRN projects.

This won't be the case when we get to binary distribution, so we'll suddenly see autolinking break for the IDE scenario.

Reply:
Auto linking changes project files. It cannot happen during the build without causing problems.

This target only checks if the files could be updated by autolinking, and warns the dev to run auto-linking.

If someone builds in VS, and _for any reason_, this check doesn't run, the *only* breaking scenario is a build break because a community module was uninstalled and auto-linking not re-run, because the previously generated files reference that missing module. The build break will be in the files that tells user what to run to fix it.

In that scenario there are _no filter files_ that changed that we could have checked for in the first place.

If you have an explicit change that fixes this, submit it. Otherwise, there's nothing to be done.